pkg/results/types.go

```diff
@@ -59,3 +75,34 @@ func (vs ViolationStore) Add(extra ViolationStore) ViolationStore {
 
 	return vs
 }
+
+// NewDefaultScanResult will initialize DefaultScanResult
+func NewDefaultScanResult(iacType, iacFilePath, iacDirPath string, totalPolicyCount int, verbose bool, violationStore ViolationStore) *DefaultScanResult {
```

Comment:
I think, Scan summary can become part of the `ViolationStore` struct itself. Scan summary is useful in all the formats. Doing so should eliminate the need to have the function `NewDefaulScanResult`.

Does it make sense?

Reply:
Yes, we would not need the DefaultScanResult struct if we add that information to ViolationStore struct. But the idea was not to make any change in the existing output.